aptos-move/move-examples/marketplace/sources/fee_schedule.move

```diff
@@ -0,0 +1,498 @@
+/// Defines the charges associated with using a marketplace, namely:
+/// * Listing rate, the units charged for creating a listing.
+/// * Bidding rate, the units per bid made by a potential buyer.
+/// * Commission, the units transferred to the marketplace upon sale.
+///
+/// Note: this is ony a schedule, a higher layer needs to implement the type of asset class
+/// and the transferring of that asset for each scheduled event.
+module marketplace::fee_schedule {
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String};
+
+    use aptos_std::type_info;
+
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::object::{Self, ConstructorRef, ExtendRef, Object};
+
+    /// FeeSchedule does not exist.
+    const ENO_FEE_SCHEDULE: u64 = 1;
+    /// The denominator in a fraction cannot be zero.
+    const EDENOMINATOR_IS_ZERO: u64 = 2;
+    /// The value represented by a fraction cannot be greater than 1.
+    const EEXCEEDS_MAXIMUM: u64 = 3;
+    /// The passed in signer is not the owner of the marketplace.
+    const ENOT_OWNER: u64 = 4;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Defines marketplace fees
+    struct FeeSchedule has key {
+        /// Address to send fees to
+        fee_address: address,
+        /// Ref for changing the configuration of the marketplace
+        extend_ref: ExtendRef,
+        /// An event stream of changes to the fee schedule
+        mutation_events: EventHandle<MutationEvent>,
```

Comment:
As a design pattern, I wonder if we should encourage events to be in a different resource now that we have resource groups